src/metabase/api/collection.clj

```diff
@@ -102,6 +102,32 @@
           (dissoc ::collection.root/is-root?)
           collection/personal-collection-with-ui-details))))
 
+(defn- shallow-tree-from-collection-id
+  "Returns only a shallow Collection in the provided collection-id, e.g.
+
+  location: /1/
+  ```
+  [{:name     \"A\"
+    :location \"/1/\"
+    :children 1}
+    ...
+    {:name     \"H\"
+     :location \"/1/\"}]
+
+  If the collection-id is nil, then we default to the root collection.
+  ```"
+  [colls]
+  (->> colls
+       (map collection/personal-collection-with-ui-details)
+       (collection/collections->tree nil)
+       (map (fn [coll] (update coll :children #(boolean (seq %)))))))
+
+(defn- location-from-collection-id-clause
+  [collection-id]
+  (if collection-id
+    [:like :location (tru "%/{0}/%" collection-id)]
+    [:not [:like :location "/%/%/"]]))
```

Comment:
This is just for selecting children and sub-children of the root, right? Could you add a comment to make that clearer, or a docstring to the function?

Reply:
Yep, will do